packages/client/hmi-client/src/components/code/tera-code.vue

```diff
@@ -0,0 +1,264 @@
+import InputText from 'primevue/inputtext';
+import { createModel } from '@/services/model';
+import { addAsset } from '@/services/project';
+import router from '@/router';
+import { RouteName } from '@/router/routes';
+import useResourceStore from '@/stores/resources';
+import FileUpload from 'primevue/fileupload';
+import TeraModelDiagram from '../model/petrinet/tera-model-diagram.vue';
+
+const props = defineProps<{
+	projectId: string;
+	assetId: string;
+}>();
+
+const emit = defineEmits(['asset-loaded']);
+
+const resourceStore = useResourceStore();
+const toast = useToastService();
+
+const codeName = ref('');
+const codeText = ref('# Paste some code here');
+const codeAsset = ref<Code>();
+const editor = ref<VAceEditorInstance['_editor'] | null>(null);
+const selectedText = ref('');
+const progress = ref(0);
+const isModalVisible = ref(false);
+const model = ref<Model>();
+const modelName = ref('');
+/**
+ * Editor initialization function
+ * @param editorInstance	the Ace editor instance
+ */
+async function initialize(editorInstance) {
+	editor.value = editorInstance;
+	editorInstance.session.selection.on('changeSelection', onSelectedTextChange);
+	editorInstance.setShowPrintMargin(false);
+}
+
+/**
+ * Event handler for selected text change in the code editor
+ */
+function onSelectedTextChange() {
+	selectedText.value = editor.value?.getSelectedText() ?? '';
+}
+
+async function saveCode() {
+	const existingCode = resourceStore.activeProjectAssets?.code.find(
+		(c) => c.name === codeName.value
+	);
+	if (existingCode?.id) {
+		console.log(existingCode);
+		console.log(codeText.value);
+		const file = new File([codeText.value], codeName.value);
+		const updatedCode = await updateCodeAsset(existingCode, file, progress);
+		if (!updatedCode) {
+			toast.error('', 'Unable to save file');
+		} else {
+			toast.success('', `File saved as ${codeName.value}`);
+			codeAsset.value = updatedCode;
+		}
+		return updatedCode;
+	}
+	const file = new File([codeText.value], codeName.value);
+	const newCodeAsset = await uploadCodeToProject(props.projectId, file, progress);
+	if (!newCodeAsset) {
+		toast.error('', 'Unable to save file');
+	} else {
+		toast.success('', `File saved as ${codeName.value}`);
+		codeAsset.value = newCodeAsset;
+		router.push({
+			name: RouteName.ProjectRoute,
+			params: {
+				pageType: AssetType.Code,
+				projectId: props.projectId,
+				assetId: codeAsset.value.id
+			}
+		});
+	}
+	return newCodeAsset;
+}
+
+async function extractModel() {
+	if (!codeAsset.value?.id) {
+		const newCodeAsset = await saveCode();
+		if (newCodeAsset && newCodeAsset.id) {
+			const extractedAmr = await codeToAMR(newCodeAsset.id);
+			if (extractedAmr) {
+				model.value = extractedAmr as Model;
+				modelName.value = extractedAmr.header.name;
+				isModalVisible.value = true;
+			}
+		}
+	}
+}
+
+async function saveModel() {
+	const modelToSave = model.value;
+	const createdModel = await createModel(modelToSave);
```

Comment:
```suggestion
	const createdModel = await createModel(model.value);
```

Reply:
Good catch, leftover from when I had to wrangle the value of modelToSave since it had `header` in it